### ISSUE TITLE
WebApp:webapp config appsettings set command causing all values to default to "false" in version 2.24.0

### ISSUE THREAD

**Issue by Bytezart** [Azure/azure-cli#18266]
**Describe the bug**
When executing the webapp config appsettings set command the result Application Settings in the WebApp Configuration value is set to false for each setting defined in the settings array.

**To Reproduce**
Executing the following will cause the all setting values to be set to false and not the value defined for the parameter.

JSON Settings Example
[
    {
        "slotSetting":  false,
        "name":  "ExampleParameter1",
        "value":  "@Microsoft.KeyVault(SecretUri=https://xxxxx.vault.azure.net/secrets/ExampleParameter1)"
    },
    {
        "slotSetting":  false,
        "name":  " ExampleParameter2",
        "value":  "@Microsoft.KeyVault(SecretUri=https://xxxxx.vault.azure.net/secrets/ExampleParameter2)"
    }
]

Command Example
az webapp config appsettings set --name $WebAppName --resource-group $ResourceGroup --settings @$settings.json

**Expected behavior**
Prior to version 2.24.0 the Application Settings in the WebApp Configuration would reflect the value defined in the JSON and would not be set to "false".

![image](https://user-images.githubusercontent.com/15107885/119928486-abc0f700-bf49-11eb-8ff1-d802b54d6bbc.png)

**Environment summary**
![image](https://user-images.githubusercontent.com/15107885/119928571-ce531000-bf49-11eb-9a2b-88ae61584451.png)

**Comment by yonzhan** [comment]
route to service team

**Comment by Kotasudhakarreddy** [comment]
Able to repro the issue.

**Comment by Kotasudhakarreddy** [comment]
Fix has been given. As a workaround please try setting an individual setting as follows.

To add a setting:
--------------------
`az webapp config appsettings set -g MyResourceGroup -n MyUniqueApp --settings WEBSITE_NODE_DEFAULT_VERSION=6.9.1`

To Make slot setting:
------------------------
`az webapp config appsettings set -g MyResourceGroup -n MyUniqueApp --slot-settings WEBSITE_NODE_DEFAULT_VERSION=6.9.1`

**Comment by mgroetan** [comment]
In light of my duplicate issue over at #18286, it’s worth noting that the suggested workaround here to directly update individual appsettings (or collections of them), rather than bundling many in a single JSON file is not always a very practical approach. That’s what we used to do before starting to use an intermediate JSON file instead. The reason being that for readability in the script, we tended to spread our 20+ appsettings on several lines, i.e. issuing the «set» command several times, without realizing that each update operation takes quite some time in itself, rather regardless of how many appsettings are handled by that single command. I suspect that there’s an implicit appservice restart for each such operation.
So if anyone currently using the JSON approach are looking for a short-term workaround, make sure you bundle as many of the appsettings as possible in one «set» command.

**Comment by Kotasudhakarreddy** [comment]
Fix has been released for this issue. please install CLI version `2.24.1` and confirm.

**Comment by mgroetan** [comment]
> Fix has been released for this issue. please install CLI version `2.24.1` and confirm.

I can confirm that this hotfix does the expected job when running locally.

**Local test:**
_Running the existing 2.2**4.0** for comparison, on a brand new appservice:_
`az --version`
```
azure-cli                         2.24.0 *

core                              2.24.0 *
telemetry                          1.0.6

Extensions:
application-insights              0.1.13

Python location 'C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\python.exe'
Extensions directory 'C:\Users\mogro\.azure\cliextensions'

Python (Windows) 3.8.9 (tags/v3.8.9:a743f81, Apr  6 2021, 13:22:56) [MSC v.1928 32 bit (Intel)]

Legal docs and information: aka.ms/AzureCliLegal


You have 2 updates available. Consider updating your CLI installation with 'az upgrade'
```

_The test data:_
`type appSettingsDummy.json`
```
[
    {
        "name": "AppSetting1",
        "slotSetting": false,
        "value": "Value1"
    },
    {
        "name": "AppSetting2",
        "slotSetting": false,
        "value": "Value2"
    },
    {
        "name": "AppSetting3",
        "slotSetting": false,
        "value": "Value3"
    }
]
```

_Importing the test data:_
`az webapp config appsettings set -n azclirepro --settings @appSettingsDummy.json`
```
[
  {
    "name": "APPINSIGHTS_INSTRUMENTATIONKEY",
    "slotSetting": false,
    "value": "e1895196-b134-40f7-837e-00cdf4cd50db"
  },
  {
    "name": "APPLICATIONINSIGHTS_CONNECTION_STRING",
    "slotSetting": false,
    "value": "InstrumentationKey=e1895196-b134-40f7-837e-00cdf4cd50db;IngestionEndpoint=https://westeurope-5.in.applicationinsights.azure.com/"
  },
  {
    "name": "ApplicationInsightsAgent_EXTENSION_VERSION",
    "slotSetting": false,
    "value": "~2"
  },
  {
    "name": "XDT_MicrosoftApplicationInsights_Mode",
    "slotSetting": false,
    "value": "default"
  },
  {
    "name": "ANCM_ADDITIONAL_ERROR_PAGE_LINK",
    "slotSetting": false,
    "value": "https://azclirepro.scm.azurewebsites.net/detectors?type=tools&name=eventviewer"
  },
  {
    "name": "AppSetting1",
    "slotSetting": false,
    "value": "False"
  },
  {
    "name": "AppSetting2",
    "slotSetting": false,
    "value": "False"
  },
  {
    "name": "AppSetting3",
    "slotSetting": false,
    "value": "False"
  }
]
```

_Upgrading to the latest CLI:_
`az upgrade`
```
This command is in preview and under development. Reference and support levels: https://aka.ms/CLI_refstatus
Your current Azure CLI version is 2.24.0. Latest version available is 2.24.1.
Please check the release notes first: https://docs.microsoft.com/cli/azure/release-notes-azure-cli
Do you want to continue? (Y/n):
Upgrading extensions
Checking update for application-insights
No updates available for 'application-insights'. Use --debug for more information.
Upgrade finished.You can enable auto-upgrade with 'az config set auto-upgrade.enable=yes'. More details in https://docs.microsoft.com/cli/azure/update-azure-cli#automatic-update
```

_Verifying the latest version:_
`az --version`
```
azure-cli                         2.24.1

core                              2.24.1
telemetry                          1.0.6

Extensions:
application-insights              0.1.13

Python location 'C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\python.exe'
Extensions directory 'C:\Users\mogro\.azure\cliextensions'

Python (Windows) 3.8.9 (tags/v3.8.9:a743f81, Apr  6 2021, 13:22:56) [MSC v.1928 32 bit (Intel)]

Legal docs and information: aka.ms/AzureCliLegal


Your CLI is up-to-date.
```

_Importing the same test data once more:_
`az webapp config appsettings set -n azclirepro --settings @appSettingsDummy.json`
```
[
  {
    "name": "APPINSIGHTS_INSTRUMENTATIONKEY",
    "slotSetting": false,
    "value": "e1895196-b134-40f7-837e-00cdf4cd50db"
  },
  {
    "name": "APPLICATIONINSIGHTS_CONNECTION_STRING",
    "slotSetting": false,
    "value": "InstrumentationKey=e1895196-b134-40f7-837e-00cdf4cd50db;IngestionEndpoint=https://westeurope-5.in.applicationinsights.azure.com/"
  },
  {
    "name": "ApplicationInsightsAgent_EXTENSION_VERSION",
    "slotSetting": false,
    "value": "~2"
  },
  {
    "name": "XDT_MicrosoftApplicationInsights_Mode",
    "slotSetting": false,
    "value": "default"
  },
  {
    "name": "ANCM_ADDITIONAL_ERROR_PAGE_LINK",
    "slotSetting": false,
    "value": "https://azclirepro.scm.azurewebsites.net/detectors?type=tools&name=eventviewer"
  },
  {
    "name": "AppSetting1",
    "slotSetting": false,
    "value": "Value1"
  },
  {
    "name": "AppSetting2",
    "slotSetting": false,
    "value": "Value2"
  },
  {
    "name": "AppSetting3",
    "slotSetting": false,
    "value": "Value3"
  }
]
```
Now the appsettings have the correct value, and the slot setting stays as "False".

_Modified test data, with one appsetting being a slot setting:_
`type appSettingsDummyV2.json`
```
[
    {
        "name": "AppSetting1",
        "value": "Value1-v2",
        "slotSetting": false
    },
    {
        "name": "AppSetting2",
        "value": "Value2-v2 - slot setting!",
        "slotSetting": true
    },
    {
        "name": "AppSetting3",
        "value": "Value3-v2",
        "slotSetting": false
    }
]
```

_Importing the modified test data:_
`az webapp config appsettings set -n azclirepro --settings @appSettingsDummyV2.json`
```
[
  {
    "name": "APPINSIGHTS_INSTRUMENTATIONKEY",
    "slotSetting": false,
    "value": "e1895196-b134-40f7-837e-00cdf4cd50db"
  },
  {
    "name": "APPLICATIONINSIGHTS_CONNECTION_STRING",
    "slotSetting": false,
    "value": "InstrumentationKey=e1895196-b134-40f7-837e-00cdf4cd50db;IngestionEndpoint=https://westeurope-5.in.applicationinsights.azure.com/"
  },
  {
    "name": "ApplicationInsightsAgent_EXTENSION_VERSION",
    "slotSetting": false,
    "value": "~2"
  },
  {
    "name": "XDT_MicrosoftApplicationInsights_Mode",
    "slotSetting": false,
    "value": "default"
  },
  {
    "name": "ANCM_ADDITIONAL_ERROR_PAGE_LINK",
    "slotSetting": false,
    "value": "https://azclirepro.scm.azurewebsites.net/detectors?type=tools&name=eventviewer"
  },
  {
    "name": "AppSetting1",
    "slotSetting": false,
    "value": "Value1-v2"
  },
  {
    "name": "AppSetting2",
    "slotSetting": true,
    "value": "Value2-v2 - slot setting!"
  },
  {
    "name": "AppSetting3",
    "slotSetting": false,
    "value": "Value3-v2"
  }
]
```

So everything seems to be in order again now, even for the last update with one of the appsettings being slot specific.
But our workaround (of not including the "slotSetting" property at all) needs to stay for the time being, as long as the 2.2**4.0** version can theoretically be in use on an ADO hosted agent, along with the previous 2.2**3**.0 version.

Note that when getting assigned a hosted agent with 2.2**3**.0, there's a side effect of turning all appsettings into slot settings when not including the "slotSetting" property in the JSON file. We can live with this compromise until the new 2.2**4.1** version is predictably assigned to hosted agents, after which we can reintroduce the "slotSetting" property once more.

**Comment by Bytezart** [comment]
I can confirm this fix is now working as intended in my test rigs. Thank you for addressing this problem quickly. It is greatly appreciated!
